concepts/eventing/performance evaluation/nats-jetstream/results_prod_HA.md

```diff
@@ -0,0 +1,332 @@
+
+```
+Events:
+  Type     Reason                  Age    From                     Message
+  ----     ------                  ----   ----                     -------
+  Normal   Scheduled               8m12s  default-scheduler        Successfully assigned kyma-system/eventing-nats-2 to shoot--kymatunas--fzn-p2-worker-k59z0-z1-5f68f-tnw8q
+  Warning  FailedAttachVolume      8m12s  attachdetach-controller  Multi-Attach error for volume "pv--f20986b4-a4a0-455b-ace2-6b91bdd4f11c" Volume is already exclusively attached to one node and can't be attached to another
+  Warning  FailedMount             6m9s   kubelet                  Unable to attach or mount volumes: unmounted volumes=[eventing-nats-js-pvc], unattached volumes=[eventing-nats-js-pvc config-volume pid kube-api-access-gn9xs]: timed out waiting for the condition
+  Warning  FailedMount             3m52s  kubelet                  Unable to attach or mount volumes: unmounted volumes=[eventing-nats-js-pvc], unattached volumes=[kube-api-access-gn9xs eventing-nats-js-pvc config-volume pid]: timed out waiting for the condition
+  Normal   SuccessfulAttachVolume  113s   attachdetach-controller  AttachVolume.Attach succeeded for volume "pv--f20986b4-a4a0-455b-ace2-6b91bdd4f11c"
+  Warning  FailedMount             97s    kubelet                  Unable to attach or mount volumes: unmounted volumes=[eventing-nats-js-pvc], unattached volumes=[config-volume pid kube-api-access-gn9xs eventing-nats-js-pvc]: timed out waiting for the condition
+```
+
+---
+
+## Test Scenario 2: K8s Cluster Node deleted during test where stream leader NATS pod was deployed
+
+### Run ID: 8/6/2022T14:24 (Duration: 5m, Event Rate: 150rps)
+
+In this test run, the k8s Node was deleted where `eventing-nats-2` Pod was deployed. The Node was deleted after 2 minutes of test execution. The `eventing-nats-2` Pod was the stream leader.
+
+**State before test run:**
+- Stream:
+  - Leader: eventing-nats-2
+  - Replicas: 1
+
+**State after test run:**
+- Stream:
+  - Leader: NA (Stream Unavailable)
+  - Replicas: NA
+
+
+**Test Results:**
+```
+-> Total Events Sent      by **Test Sender** : 10,672 (+ 45 (Failed) = 10,717)
+-> Total Events Received  by **Sink**        : 10,672
+```
+
+**Test Results Dashboard:**
+
+![](assets/ha_crash2_1.png "")
+
+**Test Sender Stats:**
+
+![](assets/ha_crash2_2.png "")
+
+**Finding:** 
+
+Deleting the K8s Node where stream leader NATS Pod was deployed resulted in the JetStream stream to be unavailable as there was only one replica of the stream. NATS stopped accepting any new events for that steam and event-publisher-proxy was failing to publish any event to NATS. The system started to work again as soon as the crashed NATS Pod was deployed on another Node and stream became available again. Therefore, a single stream replica is not enough to cater a situation of single-node failure.
```

Comment:
```suggestion
Deleting the Kubernetes Node in which the stream leader NATS Pod was deployed resulted in the JetStream stream to be unavailable because there was only one replica of the stream. NATS stopped accepting any new events for that stream and event-publisher-proxy was failing to publish any event to NATS. The system started to work again as soon as the crashed NATS Pod was deployed on another Node and the stream became available again. Therefore, a single stream replica is not enough to cater a situation of single-node failure.
```

"replica" means that it's an exact copy of something. If there's just one, isn't that the original, instead of a replica?